appengine/test/tc_sample_apps.rb

```diff
@@ -32,7 +32,7 @@ def run_app_test(dirname)
     ::Dir.chdir(::File.join(APPS_DIR, dirname)) do |dir|
       build_docker_image("", dirname) do |image|
         run_docker_daemon("-p 8080:8080 #{image}") do |container|
-          assert_cmd_output("curl -s -S http://127.0.0.1:8080/", "ruby app", 10)
+          assert_cmd_output("docker exec #{container} curl -s -S http://127.0.0.1:8080", "ruby app", 10)
```

Comment:
Ok, I see. I run the cloud build script multiple times and it always comes back with "FAILURE" status, so I automatically assumed it was the refused connection error from the tests.

After a closer look as you pointed out, I realize the connection takes multiple tries, and my failure build may be result from the final image push isn't denied.